packages/filter/src/ensure-fields.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/filter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import _ from 'lodash';
+import {Filter, FilterBuilder} from './query';
+import {AnyObject} from './types';
+
+/**
+ * Ensures that queries which apply the returned filter would always include
+ * the target fields. To undo this effect later, fields that were disabled
+ * in the original filter will be added to the pruning mask.
+ *
+ * @param targetFields - An array of fields to include
+ * @param filter - A target filter
+ * @returns A tuple containing amended filter and pruning mask
+ */
+export function ensureFields<T extends object = AnyObject>(
+  targetFields: (keyof T)[],
+  filter: Filter<T>,
+) {
+  const builder = new FilterBuilder(filter);
+  const fields = builder.build().fields;
+  if (!fields || matchesFields(targetFields, filter)) {
+    return {
+      filter: builder.build(),
+      fieldsAdded: [] as (keyof T)[],
+    };
+  }
+  const isDisablingOnly = _.size(fields) > 0 && !_.some(fields, Boolean);
```

Comment:
this is some hard to understand a suggestion could be
```suggestion
  const isDisablingOnly = Object.values(fields).every(f => !f);
```